### ISSUE TITLE
Multiple cursor behaves weird

### ISSUE THREAD

**Issue by adsick** [helix-editor/helix#1367]
This feature is described in the tutor as follows:
```
 Type C to duplicate the cursor to the next line.
```
and what I see is not quite what is written there. Idk, it is more simple to recommend you to play with it and see that it's strange but just to be more clear I will show what seems wrong to me:

## reproducing

step one, put a cursor on empty line (348 in this case)
![image](https://user-images.githubusercontent.com/24528839/147386482-bf1fa864-b13a-4322-a50f-617ae3757ceb.png)
press ```C```
![image](https://user-images.githubusercontent.com/24528839/147386488-b3454569-075c-44a0-9deb-5912f8c21123.png)
see the another cursor on line 350 and not on 349 as I would expect
press again and it will skip a line again:
![image](https://user-images.githubusercontent.com/24528839/147386502-2ad7a140-3b27-483c-8fa4-cd1436f74fb1.png)

also I see there is some very non-intuitive behavior depending on does the cursor sit on end of the line or not:
put a cursor on line 359, tap ```C``` 3 times and everything works as expected
![image](https://user-images.githubusercontent.com/24528839/147386643-b9482a4d-ef45-4702-b8b9-e1f3b0ac9d97.png)

if we do a ```gl``` to move to last char we can get this:
![image](https://user-images.githubusercontent.com/24528839/147386721-63b8293d-a1d6-4198-ae13-06f932b3da0e.png)
Helix skips two lines and puts a cursor where it can be placed

but if we go beyond last character like this:
![image](https://user-images.githubusercontent.com/24528839/147386742-478b4b59-df89-4a7d-bbe1-c2bf9296df6b.png)
and start ```C``` multiplying cursors, that's whole another deal:
![image](https://user-images.githubusercontent.com/24528839/147386757-824beee1-4780-4d84-953c-403fe5fc56fe.png)
what on Earth is that?

Just my opinion but I can't see how I can make use of multiple cursors just yet, especially in it's current state.
another thing that bothers me is that when you move leftward on the column 1 the cursor moves to the end of the previous line instead of just blocking in place (pls tell me if this behavior has its own name) and when you use multiple cursors and move that way this becomes a mess. Kinda beautiful but useless.

**Comment by archseer** [comment]
> Helix skips two lines and puts a cursor where it can be placed

Yeah so this is the expected behavior, it'll skip lines until it can position on the same visual offset. The other two behaviors (on an empty line and past the line end) seem like bugs, I'll need to do some testing.

**Comment by sudormrfbin** [comment]
> when you move leftward on the column 1 the cursor moves to the end of the previous line instead of just blocking in place

This is also expected behavior, but quite a few people seem to prefer it to stop at line bounds instead of going to next/previous line. It's currently also not configurable.

**Comment by adsick** [comment]
@sudormrfbin  I've been writing my text editor (don't mind) and there I had two types of moving left and right - one was blocking and the other was not.

**Comment by kirawi** [comment]
#768?

**Comment by glyh** [comment]
Also, when selecting empty lines, the following lines will be selected: 

[![asciicast](https://asciinema.org/a/7xbtTLjQKeYJeBpGZV7L2i1s9.svg)](https://asciinema.org/a/7xbtTLjQKeYJeBpGZV7L2i1s9)

**Comment by N4tus** [comment]
Here is another example of `C` not being intuitive. If a cursor is placed on a line break it behaves different that just on a normal character.
![helix_C_bug1](https://user-images.githubusercontent.com/33029422/160869628-c9b90bbc-bba5-4a78-9163-8aebca8d38d7.png)
![helix_C_bug2](https://user-images.githubusercontent.com/33029422/160869644-7e6970d6-b445-4eca-b32c-2fa470dfce68.png)
